packages/google_identity_services_web/example/integration_test/src/dom.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class DomDocument {}
+extension DomDocumentExtension on DomDocument {
+  DomElement createElement(String name, [Object? options]) =>
```

Comment:
The other approach is to use `@JS` and then `_createElement1(String name)` /  `_createElement2(String name, Object options)`, and then write a trampoline in `createElement`. I'm okay with this too though it might be a bit slower on Wasm backends, but I doubt this is in a hot path.

Reply:
Definitely not in a hot code path this one (used only for testing, and to be replaced by package:web as soon as it exists :) )